extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -575,16 +576,16 @@ export async function promptConnectionStringPasswordAndUpdateConnectionString(co
 				connectionString = await vscodeMssqlApi.getConnectionString(connectionDetails, true, false);
 			}
 		}
-		// set connection string to not include the password if connection info does not include password, or user chooses to not include password, or authentication type is not sql login
-		let userPassword: string | undefined;
-		if (includePassword !== constants.yesString) {
+
+		if (includePassword !== constants.yesString || !connectionInfo.password || connectionInfo.authenticationType !== 'SqlLogin') {
+			// set connection string to not include the password if connection info does not include password, 
+			// or user chooses to not include password (or if user cancels out of include password prompt), or authentication type is not SQL login
 			connectionString = await vscodeMssqlApi.getConnectionString(connectionDetails, false, false);
 
-			// Ask user to enter password if auth type is sql login and password is not saved
-			if (connectionInfo.authenticationType === 'SqlLogin' && connectionInfo.password) {
+			if (!includePassword && connectionInfo.authenticationType === 'SqlLogin') {
```

Comment:
If you do end up keeping this though - are you sure that `!includePassword` is the correct check? Because that would only be true if the user cancelled. It doesn't include the case where the user chose `No`.

Reply:
If the user chooses `No` we discussed that we would not show this prompt (manually enter the password) since that would duplicate a prompt for them.

Reply:
After adding tests I found that the only time we would ask the user to enter a password **should** be if the user does not have a connectionInfo password saved and the authentication type is SQL. 

Include password prompt can be canceled as you suggested and therefore this could also be checked.